src/uu/csplit/src/csplit.rs

```diff
@@ -427,7 +438,13 @@ impl SplitWriter<'_> {
                             input_iter.add_line_to_buffer(ln, l).is_none(),
                             "should be big enough to hold every lines"
                         );
+                    } else {
+                        // since offset_usize is for sure greater than 0
+                        // the first element of the buffer should be removed and this
+                        // line inserted to be coherent with GNU implementation
+                        input_iter.add_line_to_buffer(ln, l);
                     }
```

Comment:
A detail: I would swap the `if` and `else` blocks. It allows you to get rid of the `!` in the condition for better readability.